libs/core/src/components/sage-tooltip/sage-tooltip.tsx

```diff
@@ -0,0 +1,205 @@
+import { Component, Element, Event, Host, Prop, State, h, EventEmitter, Method, Watch } from '@stencil/core';
+import {
+  positionTooltip
+} from '../../utils/overlay';
+
+/**
+ * @slot (default) - The tooltip's target element
+ * @slot content - HTML content for the tooltip
+ */
+
+@Component({
+  tag: 'sage-tooltip',
+  styleUrl: 'sage-tooltip.scss',
+  shadow: true,
+})
+export class SageTooltip {
+  private contentEl: HTMLElement | null;
+
+  /**
+   * Reference to the Host element
+   */
+  @Element() el: HTMLDivElement;
+
+  /**
+   * Determines where the tooltip is open
+   */
+  @State() isOpen = false;
+
+  /**
+   * Content for the tooltip. If HTML is required, use the content slot
+   */
+  @Prop() content: '';
+
+  /**
+   * Id used to reference the component
+   */
+  @Prop() componentId: '';
+
+  /**
+   * Determines whether or not the tooltip have an arrow
+   */
+  @Prop() hasArrow?: boolean;
+
+
+  /**
+   * Enable this option when using the content slot
+   */
+  @Prop() htmlContent?: boolean;
+
+  /**
+   * Determines the preferred position of the tooltip
+   */
+  @Prop() placement:
+    'top'
+    | 'top-start'
+    | 'top-end'
+    | 'right'
+    | 'right-start'
+    | 'right-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'left'
+    | 'left-start'
+    | 'left-end';
+
+  /**
+   * Determines whether or not the tooltip is visible
+   */
+  @Prop({mutable: true, reflect: true}) opened = false;
+
+  // eslint-disable-next-line @stencil/no-unused-watch
+  @Watch('opened')
+  handleOpenToggle() {
+    if(this.opened) {
+      this.handleShow();
+    } else {
+      this.handleHide();
+    }
+  }
+
+  /**
+   * Determines whether or not the tooltip is disabled
+   */
+  @Prop({mutable: true}) disabled = false;
+
+  // eslint-disable-next-line @stencil/no-unused-watch
+  @Watch('disabled')
+  handleDisabled() {
+    // console.log('disabled')
+  }
+
+  /**
+   * Emitted after a tooltip is closed
+   */
+  @Event() sageTooltipHide: EventEmitter;
+
+  /**
+   * Emitted after a tooltip is shown
+   */
+  @Event() sageTooltipShow: EventEmitter;
```

Comment:
I was under the impression that the `@Event` decorator required a method directly below it to run when the event is triggered. I'm curious about what these decorators do on their own. Can you help me understand?